main.go

```diff
@@ -219,4 +222,28 @@ func serve(config *server.Config) {
 			log.Infof("%s", err)
 		}
 	}()
+
+	// watch for termination signals
+	go func() {
+		sigint := make(chan os.Signal, 1)
+		signal.Notify(sigint, os.Interrupt)    // terminal
+		signal.Notify(sigint, syscall.SIGTERM) // kubernetes
+		sig := <-sigint
+		log.Infof("shutdown signal: %s", sig)
+		signal.Stop(sigint)
+
+		grpcServer.GracefulStop()
+
+		timeout, cancel := context.WithTimeout(context.Background(), time.Second)
+		if err := httpServer.Shutdown(timeout); err != nil {
+			log.Errorf("http shutdown: %v", err)
+			os.Exit(1)
```

Comment:
It seems to me that if the http server fails to shut down timely, the Close() method will not be called which might leave some staged files. Is that expected?

Reply:
No, good catch... I had meant to remove that Exit().